### PR TITLE
Changes to Alert component to match the others

### PR DESCRIPTION
**Pull request by GeoSot**
Preview: https://deploy-preview-33402--twbs-bootstrap.netlify.app/docs/5.0/components/alerts/

### CODE REVIEW

File: site/content/docs/5.0/components/alerts.md [hunk 1]
```diff
@@ -91,23 +91,29 @@ Loop that generates the modifier classes with the `alert-variant()` mixin.
...
-Enable dismissal of an alert via JavaScript:
+Initialize elements as alerts
 
 ```js
-var alertList = document.querySelectorAll('.alert')
-alertList.forEach(function (alert) {
-  new bootstrap.Alert(alert)
-})
+  var alertList = document.querySelectorAll('.alert')
+  var alerts =  [...alertList].map(el => new bootstrap.Alert(el))
```
Comment (XhmikosR) [inline site/content/docs/5.0/components/alerts.md:100]: Please use ES5 for now since we don't use ES6 in docs yet.

**Review by alpadev — COMMENTED** [review]
Code looks ok to me. :+1: The docs could be improved IMO.
#
> You can create an alert instance with the alert constructor, for example:
```js
var myAlert = document.getElementById('myAlert')
var bsAlert = new bootstrap.Alert(myAlert)
```
This feels a bit redundant now that you've added an extra section for initialization.
#
> This makes an alert listen for click events on descendant elements which have the `data-bs-dismiss="alert"` attribute. (Not necessary when using the data-api's auto-initialization.)

Instead of removing this I guess it could be changed a bit and added to the initialization section. Maybe something like:
**Note**: For the sole purpose of dismissing an alert, it isn't necessary to initialize the component manually via the JS api. By making use of `data-bs-dismiss="alert"`, the component will be initialized automatically and properly dismissed. See the trigger section for more details.

**Review by alpadev — APPROVED** [review]
:+1: Not sure if @XhmikosR needs to approve.

File: site/content/docs/5.0/components/alerts.md [hunk 2]
```diff
@@ -147,35 +147,41 @@ Loop that generates the modifier classes with the `alert-variant()` mixin.
...
+### Initialize
 
-Enable dismissal of an alert via JavaScript:
+Initialize elements as alerts
 
 ```js
 var alertList = document.querySelectorAll('.alert')
-alertList.forEach(function (alert) {
-  new bootstrap.Alert(alert)
+var alerts = [...alertList].map(function (el) {
```
Comment (XhmikosR) [inline site/content/docs/5.0/components/alerts.md:156]: I wouldn't use spread yet. The docs are everywhere ES5

Comment (alpadev) [reply site/content/docs/5.0/components/alerts.md:156]: I'm so used to them already, I didn't even recognize that. Good catch :+1:

**Comment by tungphan995** [comment]
jeane

File: site/content/docs/5.0/components/alerts.md [hunk 3]
```diff
@@ -147,35 +147,42 @@ Loop that generates the modifier classes with the `alert-variant()` mixin.
...
 
-### Triggers
+### Initialize
 
-Enable dismissal of an alert via JavaScript:
+Initialize elements as alerts
 
 ```js
 var alertList = document.querySelectorAll('.alert')
+var alerts = []
```
Comment (XhmikosR) [inline site/content/docs/5.0/components/alerts.md:156]: I'm not sure about this here, maybe we should use `map`? Just thinking out loud here.

**Comment by GeoSot** [comment]
@XhmikosR  are we ok to proceed with this?

File: site/content/docs/5.0/components/alerts.md [hunk 4]
```diff
@@ -147,35 +147,39 @@ Loop that generates the modifier classes with the `alert-variant()` mixin.
...
 ```html
 <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
 ```
 
-Note that closing an alert will remove it from the DOM.
-
-### Methods
-
-You can create an alert instance with the alert constructor, for example:
+or on a button **outside the alert** as demonstrated above:
```
Comment (rohit2sharma95) [inline site/content/docs/5.0/components/alerts.md:174]: Not sure, it would be better to mention the `data-bs-target` here

File: js/src/alert.js [hunk 5]
```diff
@@ -111,7 +94,19 @@ class Alert extends BaseComponent {
...
 
-EventHandler.on(document, EVENT_CLICK_DATA_API, SELECTOR_DISMISS, Alert.handleDismiss(new Alert()))
+EventHandler.on(document, EVENT_CLICK_DATA_API, SELECTOR_DISMISS, function (event) {
+  if (['A', 'AREA'].includes(this.tagName)) {
+    event.preventDefault()
+  }
+
+  if (isDisabled(this)) {
+    return
+  }
```
Comment (rohit2sharma95) [inline js/src/alert.js:104]: :warning: This changes the behavior

**Review by rohit2sharma95 — APPROVED** [review]

File: js/src/alert.js (hunk 5 above)
Comment (GeoSot) [reply js/src/alert.js:104]: Is a behavior that we support on the other components. Right?
As by default dev will not use anchor elemens , either disabled (doc doesn't propose something )

File: site/content/docs/5.0/components/alerts.md (hunk 4 above)
Comment (GeoSot) [reply site/content/docs/5.0/components/alerts.md:174]:
```suggestion
or on a button **outside the alert** using the `data-bs-target` as demonstrated above:
```

File: js/src/alert.js (hunk 5 above)
Comment (rohit2sharma95) [reply js/src/alert.js:104]: That's good though, but I just wanted to mention that it was not before :slightly_smiling_face:

**Review by rohit2sharma95 — APPROVED** [review]

**Pull request merged**